lib/renderer/session.go

```diff
@@ -0,0 +1,85 @@
+package renderer
+
+import (
+	"sync"
+
+	"github.com/gorilla/websocket"
+	"github.com/leaanthony/mewn"
+	"github.com/wailsapp/wails/lib/interfaces"
+	"github.com/wailsapp/wails/lib/logger"
+)
+
+// TODO Move this back into bridge.go
+
+// session represents a single websocket session
+type session struct {
+	bindingCache []string
+	conn         *websocket.Conn
+	log          *logger.CustomLogger
+	ipc          interfaces.IPCManager
+
+	// Mutex for writing to the socket
+	lock sync.Mutex
+}
+
+// Identifier returns a string identifier for the remote connection.
+// Taking the form of the client's <ip address>:<port>.
+func (s *session) Identifier() string {
+	return s.conn.RemoteAddr().String()
+}
+
+func (s *session) sendMessage(msg string) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	if err := s.conn.WriteMessage(websocket.TextMessage, []byte(msg)); err != nil {
+		s.log.Error(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (s *session) start() {
+	s.log.Infof("Connected to frontend.")
+
+	wailsRuntime := mewn.String("../../runtime/assets/wails.js")
+	s.evalJS(wailsRuntime, wailsRuntimeMessage)
+
+	// Inject bindings
+	for _, binding := range s.bindingCache {
+		s.evalJS(binding, bindingMessage)
+	}
+
+	// FIXME: What should happen here? With multiple sessions occurring this event
+	//        seems less useful.
+	// Emit that everything is loaded and ready
+	//s.eventManager.Emit("wails:ready")
```

Comment:
People rely on this event so we still need to keep it. It's really just an indicator that the frontend is loaded so the backend can use the frontend runtime. Suggest we emit this on first load. I guess it gets tricky as there's no concept of multiple sessions in the backend.

Reply:
No problem. I think I know where to put it instead.  Doesn't make sense to me to emit this everytime a frontend connection is made? Does it need to happen when a connection is made? Or can it only happen once when the Bridge is initialized and ready?

Reply:
Perhaps a new event like `wails:bridge-session` for everytime a session is initiated / rejoined?

Reply:
I can easily output it only after the first session connects.  Would that be acceptable?

Reply:
Ok this has been corrected. It will now be emitted but only by the first session spawned by the bridge.

Reply:
> Perhaps a new event like `wails:bridge-session` for everytime a session is initiated / rejoined?

This is a great idea, passing the identifier with it.